Update wording in web security docs (#4136)
jupyterhub/jupyterhub: docs/source/reference/websecurity.md

# Security Overview

> The **Security Overview** section helps you learn about: the semi-trusted user;

Reviewer: I think the bulleted list delivers this info better

Author: I've tried to make the items on the list reference links to the corresponding headings or even turning it to a table of contents but couldn't get to do it, then thought about this alternative... I agree the bulleted list is simpler and makes it more rapidly comprehensible though.

the available mitigations to protect untrusted users from each other;
the value of periodic security audits.

This overview also helps you obtain a deeper understanding of how JupyterHub
works, as well as its design with respect to web security.

## Semi-trusted and untrusted users

JupyterHub is designed to be a _simple multi-user server for modestly sized
groups_ of **semi-trusted** users. Its design reflects serving semi-trusted

> users, however, JupyterHub is not necessarily unsuitable for serving **untrusted** users.

Reviewer: I prefer the `While...` construction here. Can you discuss why you think the `however` works better?

Author: The phrase sounds confusing to me, so I added the `however` so it could bring an idea of opposition between semi-trusted and untrusted user's cases.

Reading it again and again, maybe my discomfort is on `is not necessarily unsuitable`. It could be, then, `While the design reflects serving semi-trusted users, JupyterHub can also be suitable for serving **untrusted** users.`?

The **Security Overview** section helps you learn about:

- the semi-trusted user
- the available mitigations to protect untrusted users from each other
- the value of periodic security audits

This overview also helps you obtain a deeper understanding of how JupyterHub

> works, as well as its design with respect to web security.

Reviewer: Can I ask why this item is moved out of the bullets? It makes sense to me for this to be the first item, rather than the last.

Author: It was due to me trying to use reference links in the topics and headings before. Web security's design encompasses all the page, so I've put it this way, but since I couldn't use reference links to the headings, it makes sense to be put as a first item, yes.

## Semi-trusted and untrusted users

JupyterHub is designed to be a _simple multi-user server for modestly sized
groups_ of **semi-trusted** users. While the design reflects serving
semi-trusted users, JupyterHub can also be suitable for serving **untrusted** users.

> That way, using JupyterHub with **untrusted** users means more work by the

Reviewer: Perhaps a better connection:

```suggestion
As a result, using JupyterHub with **untrusted** users means more work by the
```
